pkg/collector/corechecks/snmp/internal/report/report_memory_usage.go

```diff
@@ -0,0 +1,300 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package report
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/snmp/internal/checkconfig"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/snmp/internal/valuestore"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// EvaluatedSampleDependencies set of supported memory usage metrics
+var EvaluatedSampleDependencies = map[string]bool{
+	"memory.usage": true,
+	"memory.used":  true,
+	"memory.total": true,
+	"memory.free":  true,
+}
+
+func (ms *MetricSender) tryReportMemoryUsage(scalarSamples map[string]MetricSample, columnSamples map[string]map[string]MetricSample) error {
+	if ms.hasScalarMemoryUsage(scalarSamples) {
+		return ms.trySendScalarMemoryUsage(scalarSamples)
+	}
+
+	return ms.trySendColumnMemoryUsage(columnSamples)
+}
+
+// hasScalarMemoryUsage is true if samples store contains at least one memory usage related metric
+func (ms *MetricSender) hasScalarMemoryUsage(scalarSamples map[string]MetricSample) bool {
+	_, scalarMemoryUsageOk := scalarSamples["memory.usage"]
+	_, scalarMemoryUsedOk := scalarSamples["memory.used"]
+	_, scalarMemoryTotalOk := scalarSamples["memory.total"]
+	_, scalarMemoryFreeOk := scalarSamples["memory.free"]
```

Comment:
🥜  Maybe we can just not have `memory.usage` here and avoids the need to call `trySendScalarMemoryUsage`
If, so, maybe the function should be instead `shouldSendScalarMemoryUsage`

That will also remove the need of those lines:

```
	_, scalarMemoryUsageOk := scalarSamples["memory.usage"]
	if scalarMemoryUsageOk {
		// memory usage is already sent through collected metrics
		return nil
	}
```

Reply:
`shouldSendScalarMemoryUsage` in my opinion is not super clear. should we send it because it was not sent yet ? or because we have enough metrics to send it ? 

Would it be just moving around part of this function logic, in the end ?

Reply:
My main point is that `hasScalarMemoryUsage` currently includes "memory.usage", that is a case that is skipped later with:

```
	_, scalarMemoryUsageOk := scalarSamples["memory.usage"]
	if scalarMemoryUsageOk {
		// memory usage is already sent through collected metrics
		return nil
	}
```

Not having "memory.usage" in `hasScalarMemoryUsage` in the first place might avoid having this code.

That said, I this is more a nitpick, so, it's fine if you think the currently way is prefered.